types/listlike_partial_deserializing_iterator.hh

```diff
@@ -28,6 +28,12 @@
 int read_collection_size(bytes_view& in, cql_serialization_format sf);
 bytes_view read_collection_value(bytes_view& in, cql_serialization_format sf);
 
+template <FragmentedView View>
+int read_collection_size(View& in, cql_serialization_format);
+
+template <FragmentedView View>
+View read_collection_value(View& in, cql_serialization_format);
+
```

Comment:
How does it work? These are templates, if they are defined in a source file, then linkage may break if used with the wrong types. If they are defined in a header file, then this is unneeded.

Is it just needed as a forward declaration?

Reply:
These templates are explicitly instantiated in types.cc, so magically enough it works (no compile/linkage errors, even without an extern template decl - same as collection type deserialize templates).

Reply:
ok.